extensions/amp-story/1.0/amp-story-system-layer.js

```diff
@@ -466,11 +469,40 @@ export class SystemLayer {
       this.getShadowRoot().removeAttribute(HAS_INFO_BUTTON_ATTRIBUTE);
     }
 
+    this.maybeBuildAttribution_();
+
     this.getShadowRoot().setAttribute(MESSAGE_DISPLAY_CLASS, 'noshow');
     this.getShadowRoot().setAttribute(HAS_NEW_PAGE_ATTRIBUTE, 'noshow');
     return this.getRoot();
   }
 
+  /** @private */
+  maybeBuildAttribution_() {
+    if (this.viewer_ && this.viewer_.getParam('attribution') === 'auto') {
+      this.systemLayerEl_.querySelector(
+        '.i-amphtml-story-attribution-logo'
+      ).src =
+        getStorySrcAttribute(this.parentEl_, 'entity-logo-src') ??
+        getStorySrcAttribute(this.parentEl_, 'publisher-logo-src');
+
+      const anchorEl = this.systemLayerEl_.querySelector(
+        `.${escapeCssSelectorIdent(ATTRIBUTION_CLASS)}`
+      );
+
+      anchorEl.href =
+        getStorySrcAttribute(this.parentEl_, 'entity-url') ??
+        getSourceOrigin(Services.documentInfoForDoc(this.parentEl_).sourceUrl);
+
+      this.systemLayerEl_.querySelector(
+        '.i-amphtml-story-attribution-text'
+      ).textContent =
+        this.parentEl_.getAttribute('entity') ??
```

Comment:
Here and above, `??` will give you the left operand if it's an empty string. You want to use `||` to make sure that if `entity` is empty it doesn't use it.

Reply:
Oh, good catch.